proc_linewidth.py

```diff
@@ -9,57 +9,71 @@
 from symfit.contrib.interactive_guess import InteractiveGuess
 import numpy as np
 
-fl = fl_mod()
-for searchstr,exp_type,postproc in [
-        ["201118_1mM4AT",'ESR','ESR_linewidth']
+B_center = Parameter('B_center', value=-0.2)
+sigma = Parameter('sigma', value=3)
+R = Parameter('R', value=7.0)
+A = Parameter('A', value=3e4)
+C = Parameter('C', value=0)
+B = Variable('B')
+y_var = Variable('y')
+conc_list = []
+R_list = []
+sigma_list = []
+for searchstr,exp_type,postproc,thisguess,interactive,concentration in [
+        ("201118_10mM4AT",'francklab_esr/alex','ESR_linewidth',
+            { # here, I entered based on the next, and then copied and pasted the result
+                A: 952.059548814867,
+                B_center: 0.645245462402653,
+                R: 1.1452099193249357,
+                sigma: 0.7520327837593248,
+                },
+            False,
+            10e-3
+            ),
+        ("201118_1mM4AT",'francklab_esr/alex','ESR_linewidth',
+            {
+
+                A:         4e2,
+                B_center:  4.220642e-01,
+                R:         3.617994e-01,
+                sigma:     8.702270e-01,
+                },
+            False,
+            1e-3
+            )
         ]:
     d = find_file(searchstr + '.DSC', exp_type=exp_type, postproc=postproc,
                   lookup=postproc_dict)
-    fl.next('linewidth for 10mM 4AT')
-    fl.plot(d)
+    plt.figure()
+    plt.title('linewidth for 10mM 4AT')
+    plot(d)
     d = d['$B_0$':(-9, 9)]
+    plot(d, '--', alpha=0.5, linewidth=4)
     d.setaxis('$B_0$', lambda x: x+1) # for a positive B_center, b/c the interactive guess doesn't deal well with negative parameters
-    fl.next('sliced')
-    fl.plot(d)
     s_integral =d.C.run_nopop(np.cumsum, '$B_0$')
-    #fl.next('absorbance')
-    #fl.plot(s_integral)
     #{{{fitting with voigt
-    B_center = Parameter('B_center', value=-0.2)
-    sigma = Parameter('sigma', value=3)
-    R = Parameter('R', value=7.0)
-    A = Parameter('A', value=3e4)
-    C = Parameter('C', value=0)
-    B = Variable('B')
-    y_var = Variable('y')
     if not os.path.exists('dVoigt.pickle'):
         with open('dVoigt.pickle','wb') as fp:
             # cache the expression, which takes some time to generate
             print("no pickle file found -- generating")
             z = ((B-B_center) + s.I*R)/sigma/s.sqrt(2)
             faddeeva = s.simplify(s.exp(-z**2) * s.erfc(-s.I*z))
             voigt = A*s.re(faddeeva)/sigma/s.sqrt(2*s.pi)
+            voigt *= sigma * R # so adjusting linewidth doesn't change amplitude
             voigt = voigt.simplify()
-            dVoigt = voigt.diff(B).simplify()
+            # add real below b/c lambdify was giving complex answer
+            dVoigt = s.re(s.re(voigt.diff(B)).simplify())
             pickle.dump(dVoigt,fp)
```

Comment:
so will this generate a pickle file each time I run it hence I will need to delete the pickle file every time before I run it?

Reply:
No!  You only need to regenerate the pickle if you mess with the expression. The final expression takes a while to calculate, which is why I cache it in a pickle file